alg/viewshed.cpp

```diff
@@ -280,12 +282,19 @@ bool Viewshed::calcOutputExtent(int nX, int nY)
     oOutExtent.xStop = GDALGetRasterBandXSize(pSrcBand);
     oOutExtent.yStop = GDALGetRasterBandYSize(pSrcBand);
 
-    if (nX < 0 || nX >= oOutExtent.xStop || nY < 0 || nY >= oOutExtent.yStop)
+    if (!oOutExtent.containsY(nY))
     {
         CPLError(CE_Failure, CPLE_AppDefined,
-                 "The observer location falls outside of the DEM area");
+                 "Observer position above or below the raster "
+                 "not currently supported");
         return false;
     }
+    if (!oOutExtent.contains(nX, nY))
+    {
+        CPLError(CE_Warning, CPLE_AppDefined,
```

Comment:
do we need to issue a warning if this is nominally supported ?

Reply:
I think so since it's a change in behavior from what exists. I don't think it hurts anything. It could be removed later.